### PR TITLE
 Set bucket capacity to threshold + 1 on error and always log header

### PR DESCRIPTION
**Pull request by the-gigi**
[HDO-689 Distributed API usage management for shopify](https://myhelix.atlassian.net/browse/HDO-689)

### CODE REVIEW

**Review by ahammond — APPROVED** [review]

File: goshopify.go [hunk 1]
```diff
@@ -241,12 +241,16 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
...
 
 	if callLimit, ok := resp.Header[callLimitHeader]; ok {
+		logMessage := fmt.Sprintf("%s: %s", callLimitHeader, callLimit[0])
 		// Extract the current bucket capacity from the call limit response header
 		bucketCapacity, err = strconv.Atoi(strings.Split(callLimit[0], "/")[0])
 		if err != nil {
-			return err
+			logMessage = "Unable to parse the call limit header. " + logMessage
+			// set bucket capacity to the threshold + 1 to throttle requests until call limit header can be parsed
+			bucketCapacity = c.app.BucketThreshold + 1
```
Comment (MayaLi) [inline goshopify.go:250]: `c.app.BucketThreshold` is never set. So bucketCapacity will just be 1 when no header exists. correct?

Comment (MayaLi) [reply goshopify.go:250]: Also, `c.app.HeavyLoadDelay` is never set. so line 230 is never actually going to take effect. Also please add a log line when doing sleep there

**Review by MayaLi — COMMENTED** [review]
@the-gigi  not sure what your plans are but the changes you have doesn't actually have any effect on backing off shopify requests

File: goshopify.go (hunk 1 above)
Comment (the-gigi) [reply goshopify.go:250]: the c.appBucketThreshold and c.app.HeavyLoadDelay are set from the outside by clients. If the clients don't set them then no auto-throttling is happening (same behavior as before). This is by design.  See the PRs for all the services that use go-shopify
https://github.com/myhelix/shopify-app/pull/74
https://github.com/myhelix/order/pull/94
https://github.com/myhelix/customer/pull/268
https://github.com/myhelix/fulfillment-workflow/pull/110

**Pull request merged**